pkg/proxy/topology.go

```diff
@@ -150,20 +150,8 @@ func deprecatedTopologyFilter(nodeLabels map[string]string, topologyKeys []strin
 	return filteredEndpoints
 }
 
-// filterEndpointsInternalTrafficPolicy returns the node local endpoints based
-// on configured InternalTrafficPolicy.
-//
-// If ServiceInternalTrafficPolicy feature gate is off, returns the original
-// EndpointSlice.
-// Otherwise, if InternalTrafficPolicy is Local, only return the node local endpoints.
-func filterEndpointsInternalTrafficPolicy(internalTrafficPolicy *v1.ServiceInternalTrafficPolicyType, endpoints []Endpoint) []Endpoint {
```

Comment:
@maplain I simplified this to just FilterLocalEndpoint since we previously passed in `internalTrafficPolicy` when it implemented `PreferLocal`. Now that it's just Local or Cluster, this can just check the local case.